### PR TITLE
ported SSM

### PR DESCRIPTION
**Pull request by rayoogmy**

### CODE REVIEW

File: src/integrationtest/resources/templates/instances.yml [hunk 1]
```diff
@@ -92,13 +98,25 @@ Resources:
...
+            - 10
+            - "|Ref|appVersion"
         root: true
         runtime: Docker
         dockercfg:
           'https://my-private-registry':
             auth: "my-secret-string"
             email: "test@example.com"
         environment:
+          SOME_VALUE_FROM_TEXTFILE: "|Ref|textFileParameter"
```
Comment (steveparker-1984) [inline src/integrationtest/resources/templates/instances.yml:119]: Text file?

File: src/integrationtest/resources/templates/instances.yml [hunk 2]
```diff
@@ -108,3 +126,5 @@ Resources:
                 b: !Ref someParameterB
                 c: "foo"
         kms_encrypted_value: "|Ref|kmsEncryptedValue"
+        ssm_plain_value: "|Ref|ssmPlainValue"
```
Comment (steveparker-1984) [inline src/integrationtest/resources/templates/instances.yml:129]: What's going on here?

File: src/main/python/cfn_sphere/aws/ssm.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    def __init__(self, region='eu-west-1'):
+        self.client = boto3.client('ssm', region_name=region)
+
+    def get_parameter(self, name, with_decryption=True):
+        try:
+            return self.client.get_parameter(Name=name, WithDecryption=with_decryption)['Parameter']['Value']
+        except (Boto3Error, ClientError) as e:
+            raise CfnSphereBotoError(e)
+
+if __name__ == "__main__":
```
Comment (steveparker-1984) [inline src/main/python/cfn_sphere/aws/ssm.py:17]: Is this used?

File: src/main/python/cfn_sphere/stack_configuration/parameter_resolver.py [hunk 4]
```diff
@@ -89,49 +96,72 @@ def resolve_parameter_values(self, stack_name, stack_config, cli_parameters=None
         stack_outputs = self.cfn.get_stacks_outputs()
 
         for key, value in stack_config.parameters.items():
+            resolved_parameters[key] = self.resolve_parameter_value(key, value, stack_name, stack_config, stack_outputs)
 
-            if isinstance(value, (list, TransformList)):
```
Comment (steveparker-1984) [inline src/main/python/cfn_sphere/stack_configuration/parameter_resolver.py:93]: The diff from here down isn't really understandable, what's happening here?

File: src/main/python/cfn_sphere/util.py [hunk 5]
```diff
@@ -180,6 +205,9 @@ def get_git_repository_remote_url(working_dir):
...
             return None
 
+def get_resources_dir():
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    return os.path.realpath(os.path.join(script_dir, "../../resources"))
 
 if __name__ == "__main__":
-    pass
\ No newline at end of file
+    print(get_resources_dir())
```
Comment (steveparker-1984) [inline src/main/python/cfn_sphere/util.py:213]: Required?

File: src/unittest/python/aws_tests/ssm_tests.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+try:
+    from unittest2 import TestCase
+    from mock import patch
+except ImportError:
+    from unittest import TestCase
+    from mock import patch
+from cfn_sphere.aws.ssm import SSM
+class SSMTests(TestCase):
```
Comment (steveparker-1984) [inline src/unittest/python/aws_tests/ssm_tests.py:8]: Please add spacing

File: src/integrationtest/resources/templates/instances.yml (hunk 1 above)
Comment (rayoogmy) [reply src/integrationtest/resources/templates/instances.yml:119]: Don't really know - merged in from cfn-sphere. Looks like it's got something to do with this https://github.com/zalando-stups/taupage.

File: src/integrationtest/resources/templates/instances.yml (hunk 2 above)
Comment (rayoogmy) [reply src/integrationtest/resources/templates/instances.yml:129]: It's testing the SSM keyword as far as I can see?

File: src/main/python/cfn_sphere/aws/ssm.py (hunk 3 above)
Comment (rayoogmy) [reply src/main/python/cfn_sphere/aws/ssm.py:17]: this is now added in?

File: src/main/python/cfn_sphere/stack_configuration/parameter_resolver.py (hunk 4 above)
Comment (rayoogmy) [reply src/main/python/cfn_sphere/stack_configuration/parameter_resolver.py:93]: upstream have refactored and split `resolve_parameter_values` into `resolve_parameter_values` looping and callling `resolve_parameter_value`.

File: src/main/python/cfn_sphere/util.py (hunk 5 above)
Comment (rayoogmy) [reply src/main/python/cfn_sphere/util.py:213]: No, but this is to keep inline with upstream to make the next merge easier.

File: src/unittest/python/aws_tests/ssm_tests.py (hunk 6 above)
Comment (rayoogmy) [reply src/unittest/python/aws_tests/ssm_tests.py:8]: This is merged from upstream so if we change this we'll need to change this every time... :( 

(I would love to insert more spacing by the way)

**Review by steveparker-1984 — DISMISSED** [review]

**Review by steveparker-1984 — APPROVED** [review]

**Pull request merged**